.github/workflows/check-catalyst.yaml

```diff
@@ -31,8 +37,10 @@ jobs:
         run: |
           # TODO: reenable kokkos builds
           COMPILER_LAUNCHER="" \
+          C_COMPILER=$(which ${{ needs.constants.outputs[format('c_compiler.{0}', matrix.compiler)] }}) \
```

Comment:
What does `{0}` mean? Does it need to be updated for building with both `clang` and `gcc` in the future?

Reply:
That's how most `format` like functions do argument substitution, for example the Python one: https://docs.python.org/3/library/string.html#format-examples

Reply:
Ah, right!